pkg/aggregator/aggregator.go

```diff
@@ -69,23 +71,26 @@ func (s *referenceWalker) walkSchema(schema *spec.Schema) {
 		return
 	}
 	schema.Ref = s.walkRef(schema.Ref)
-	for _, v := range schema.Definitions {
+	for k, v := range schema.Definitions {
 		s.walkSchema(&v)
+		schema.Definitions[k] = v
 	}
-	for _, v := range schema.Properties {
+	for k, v := range schema.Properties {
 		s.walkSchema(&v)
+		schema.Properties[k] = v
 	}
-	for _, v := range schema.PatternProperties {
+	for k, v := range schema.PatternProperties {
 		s.walkSchema(&v)
+		schema.PatternProperties[k] = v
```

Comment:
Can we do it consistently. e.g. for cases like this (and above):

```
for k, _ := range schema.PatternProperties {
  s.walkSchema(&schema.PatternProperties[k])
}
```

Reply:
We can't, because `Schema.PatternProperties` is a map and you can't get a pointer to a map entry. 

I could change the rest to make them consistent with all the above, but that would just add unnecessary lines of code and would also introduce copying of the array values, which is unnecessary.

Reply:
make sense.